Feature/user login

jenil123 opened the pull request:
Added code for user login service
Added code for forgot password feature

bhautikgosai96 commented:
Hi Jenil,

Remove variable attempts as you are not using it and also parameters HttpServletRequest request, HttpServletResponse response from loginUser PostMapping as they are also not used.

bhautikgosai96 reviewed (changes requested):
Hi Jenil,

- Remove variable attempts as you are not using it and also parameters HttpServletRequest request, HttpServletResponse response from loginUser PostMapping as they are also not used.
- In ResetPassword model request class add annotation @DaTa so we don't need to write getter/setters and if you need to write constructor add annotation @AllArgsConstructor and remove getter/setters/constructor. Do same changes in userLogin model request class also.
- As you have made request model, same way make response model, as your are just returning string, make message for particular response message and other fields if require. you can refer same in my code.
- Remove System.out.println() from userservice class.

jenil123 commented:
> Hi Jenil,
> 
> * Remove variable attempts as you are not using it and also parameters HttpServletRequest request, HttpServletResponse response from loginUser PostMapping as they are also not used.
> * In ResetPassword model request class add annotation @DaTa so we don't need to write getter/setters and if you need to write constructor add annotation @AllArgsConstructor and remove getter/setters/constructor. Do same changes in userLogin model request class also.
> * As you have made request model, same way make response model, as your are just returning string, make message for particular response message and other fields if require. you can refer same in my code.
> * Remove System.out.println() from userservice class.


Hi Bhautik,
As per the suggestions I have changed the code for request package and added a response package for meaningful responses. Please review the same. 

Thanks 
Jenil

ankit2204 commented on src/main/java/com/pluang/OnboardingUserService/service/UserService.java:
```diff
@@ -1,23 +1,188 @@
-    public User getUser(Integer id) {
-        User user = userRepo.findById(id).orElse(new User());
-        return user;
+    public UserResponse isCorrect(UserLogin loginInfo) throws NoSuchAlgorithmException {
```
We should use a better name for this function like validateUserLogin.

ankit2204 commented on src/main/java/com/pluang/OnboardingUserService/service/UserService.java:
```diff
@@ -1,23 +1,188 @@
+            return  1;
+        }
+    }
+    public boolean checkIfBlocked(String mobNumber)
```
This is a good implementation, but we should not keep track of these attributes in memory, by doing that we are making our service stateful.
Problem with that is, suppose we have multiple instances of user service, every request will be routed to a different instance and our checks will fail because other instances are not sharing this information. 

Correct way to implement these use cases will be using Spring Security, Go through the documentation of spring security.

aditya-jha commented on src/main/java/com/pluang/OnboardingUserService/entities/User.java:
```diff
@@ -1,13 +1,17 @@
 import java.sql.Timestamp;
 
 @Data
+@AllArgsConstructor
```
why do you need all args constructor?

aditya-jha commented on src/main/java/com/pluang/OnboardingUserService/controller/UserController.java:
```diff
@@ -1,31 +1,44 @@
-        User user = userService.getUser(id);
-        return user;
+    @PostMapping("/loginUser")
+    public UserResponse postUserInfo(@RequestBody UserLogin loginInfo) throws NoSuchAlgorithmException {
```
postUserInfo sounds more like you are posting details of a new user. Fix the name

jenil123 commented:
Hi Aditya and Ankit, 
I have changed the code as per the suggestions. I have used spring security for checking if user is logged In or not. And used a separate table for keeping track of unsuccessful login attempts.
